test/index.test.js

```diff
@@ -47,397 +25,92 @@ describe('index test', () => {
-                executor: [{
-                    name: 'k8s',
-                    options: k8sPluginOptions
-                }]
-            });
-
-            const executorKubernetes = executor.k8s;
-
-            k8sOptions.ecosystem = {};
-
-            assert.deepEqual(executorKubernetes.constructorParams, k8sOptions);
-        });
-
-        it('defaults to an empty object when config does not exist', () => {
-            const error = new Error('No executor config passed in.');
-
-            try {
-                executor = new Executor();
-            } catch (err) {
-                assert.deepEqual(err, error);
-
-                return;
-            }
-            assert.fail();
-        });
-
-        it('throws an error when the executor config does not exist', () => {
-            const error = new Error('No executor config passed in.');
-
-            try {
-                executor = new Executor({ ecosystem });
-            } catch (err) {
-                assert.deepEqual(err, error);
-
-                return;
-            }
-            assert.fail();
-        });
-
-        it('throws an error when the executor config is not an array', () => {
-            const error = new Error('No executor config passed in.');
-
-            try {
-                executor = new Executor({
-                    ecosystem,
-                    executor: {
-                        name: 'k8s',
-                        options: k8sPluginOptions
-                    }
-                });
-            } catch (err) {
-                assert.deepEqual(err, error);
-
-                return;
-            }
-            assert.fail();
-        });
-
-        it('throws an error when the executor config is an empty array', () => {
-            const error = new Error('No executor config passed in.');
-
-            try {
-                executor = new Executor({
-                    ecosystem,
-                    executor: []
-                });
-            } catch (err) {
-                assert.deepEqual(err, error);
-
-                return;
-            }
-            assert.fail();
-        });
-
-        it('throws an error when no default executor is set', () => {
-            const error = new Error('No default executor set.');
-
-            try {
-                executor = new Executor({
-                    ecosystem,
-                    executor: [{
-                        name: 'DNE'
-                    },
-                    {
-                        name: 'DNE2',
-                        options: k8sPluginOptions
-                    }]
-                });
-            } catch (err) {
-                assert.strictEqual(err.message, error.message);
+    after(() => {
+        mockery.disable();
+    });
 
-                return;
-            }
-            assert.fail();
+    describe('construction', () => {
+        it('constructs the executor', () => {
+            assert.ok(executor);
```

Comment:
To properly test construction occurred, you probably should do more than test truthiness: `assert.instanceOf(executor, Executor)`